### DISCUSSION TITLE
Mission Pinball integration with VPinMame or Visual Pinball X

### DISCUSSION THREAD

**Discussion by bernarma** [missionpinball/mpf#1818] [Q&A]
I am in the process of creating a real machine but I am using VPX as the basis for the virtual design. Rather than code rules into VPX and then have to transfer them to Mission Pinball in the future, I was wondering if something exists out of the box that can be used?

I have seen the MPF bridge project which publishes a COM object to expose the system but I have been unsuccessful to date getting the ActiveX object created (I have tried running VPX as admin as well to no avail).

Do you have any suggestions or how this could be achieved? Thanks.

**Comment by kylenahas** [comment]
I was successful in connecting MPF and VPX in my “Morgar” project, however it may be a few years out of date now. 

https://github.com/kylenahas/morgar

There was a specific sequence which things had to be started, from the readme:

Morgar is currently playable via Visual Pinball X (VPX). To do so, first register the MPF-VPCOM bridge, start mpf both, open VPX as Administrator, and select the "Gorgar MPF.vpx" file within the /vpx/ directory. Pressing the "1" key starts a game, "Enter" plunges, and Left/Right "Shift" flips.

**Reply by bernarma** [reply]
Thanks for the reply - much appreciated.

To troubleshoot things a little further I found that when using the Python version installed from the Windows Store is used, I would get issues when the COM object was instantiated via CreateObject. I have removed that version and replaced it with the official version directly from the Python Foundation. I have tested that this works using a test Python class I created to validate it working. I'll now take a look at getting the MPF Controller class working next and report back if I have any issues.

I think it is a fantastic framework hence why I am persisting in getting it working. The complexity I feel my ruleset will have will not be maintainable using procedural programming especially when multiple things are happening at the same time. I know this could also blowout, but I prefer using a more structured approach anyhow.

**Reply by bernarma** [reply]
Got it all working. I did have to run as administrator for the first run but after that, I can run VPX as a normal user. Will need to focus a little more on the performance tuning aspect of things as it is definitely more resource intensive as opposed to using vPinMame only.
